### ISSUE TITLE
Computation time for i3d features

### ISSUE THREAD

**Issue by 1980x** [v-iashin/video_features#120]
Hi. Thanks for awesome work. 
I am not able to extract visual features in an efficient way. Its taking too much time even on GPUs. I am extracting visual features on 278 videos of 4-5 minutes duration by dividing into 16 parallel subsets simultaneously on 4GPUs of 24GB RAM. It has extracted features for only 60 videos in 24 hours.  Can you suggest an efficient way for the same?

Thank you.

**Comment by v-iashin** [comment]
Hi, sorry to hear about that. 

Can you give me a bit more information? 

Did you try to divide into 4 subsets (1 per GPU, instead of running 4? on one GPU)?

It is faster with `streams=rgb`?

Do you see your GPU being 100% utilized?

**Comment by 1980x** [comment]
Yes. I also divided into 4 subsets running on each of the gpus along 4 more ubsets on each gpu. And gpu utilization was 100%. 
Is it recommended to use multiple subsets per GPU to utilize the GPU memory?

I need both rgb and flow feature so did not try separately. Do you recommend this?

**Comment by 1980x** [comment]
One more issue is that out of 16 parallel jobs submitted, it has computed features for first 4 jobs running one of each of 4 GPUs. Reaming 12 jobs, which were computing features, did not save the feature even though job was running full time. All jobs were writing the output features to same folder. It means if there is single GPU, I can't run multiple jobs to process more number of videos? Am I right? Can you suggest a solution as it is really taking lots of time.

**Comment by v-iashin** [comment]
Try using one process per GPU. Did it improve speed?
